pkg/ext-proc/backend/provider.go

```diff
@@ -47,11 +50,11 @@ func (p *Provider) AllPodMetrics() []*PodMetrics {
 }
 
 func (p *Provider) UpdatePodMetrics(pod Pod, pm *PodMetrics) {
-	p.podMetrics.Store(pod, pm)
+	p.podMetrics.Store(pod.Name, pm)
```

Comment:
to be safe, we should use NamespacedName

Reply:
can I do that as a follow up? the informer lists pods only from the namespace of the inferencePool, so this is safe until we change to support multiple pools.

Reply:
sg

Reply:
Thanks, added a todo.